File: LibsodiumSealedCryptoboxEncryptionString/LibsodiumSealedCryptoboxEncryptionString.cs
using System;
using System.Security.Cryptography;
using Sodium;

using System.IO;

public static class Program {
	public static void Main() {
		Console.WriteLine("Libsodium sealed crypto box hybrid string encryption");
		// you need this library: https://www.nuget.org/packages/Sodium.Core/ version 1.2.3
		string plaintext = "The quick brown fox jumps over the lazy dog";

		// for encryption you need the public key from the one you are sending the data to (party B)
		string publicKeyBBase64 = "jVSuHjVH47PMbMaAxL5ziBS9/Z0HQK6TJLw9X3Jw6yg=";
		// convert the base64 encoded keys
		byte[] publicKeyB = Base64Decoding(publicKeyBBase64);
		// encrypt
		string ciphertextBase64 = SealedCryptoboxEncryptionToBase64(publicKeyB, plaintext);
		Console.WriteLine("\n* * * encryption * * *");
		Console.WriteLine("all data are in Base64 encoding");
		Console.WriteLine("publicKeyB:  " + publicKeyBBase64);
		Console.WriteLine("ciphertext:  " + ciphertextBase64);
		Console.WriteLine("output is (Base64) nonce : (Base64) ciphertext");

		// for decryption you need your private key (privateKeyB)
		// received ciphertext
		string ciphertextReceivedBase64 = ciphertextBase64;
		// own private key
		string privateKeyBBase64 = "yNmXR5tfBXA/uZjanND+IYgGXlrFnrdUiUXesI4fOlM=";
		// convert the base64 encoded keys
		byte[] privateKeyB = Base64Decoding(privateKeyBBase64);

		// decrypt
		string decryptedtext = SealedCryptoboxDecryptionToBase64(privateKeyB, ciphertextReceivedBase64);
		Console.WriteLine("\n* * * decryption * * *");
		Console.WriteLine("ciphertext:  " + ciphertextReceivedBase64);
		Console.WriteLine("input is (Base64) nonce : (Base64) ciphertext");
		Console.WriteLine("privateKeyB: " + privateKeyBBase64);
		Console.WriteLine("decrypt.text:" + decryptedtext);
	}

	static string SealedCryptoboxEncryptionToBase64(byte[] publicKey, string plaintext) {
		var ciphertext = SealedPublicKeyBox.Create(plaintext, publicKey);
		return Base64Encoding(ciphertext);
	}

	static string SealedCryptoboxDecryptionToBase64(byte[] privateKey, string ciphertextReceivedBase64) {
		byte[] ciphertext = Base64Decoding(ciphertextReceivedBase64);
		KeyPair keypair = PublicKeyBox.GenerateKeyPair(privateKey);
		var decrypted = SealedPublicKeyBox.Open(ciphertext, keypair);
		return System.Text.Encoding.UTF8.GetString(decrypted, 0, decrypted.Length);
	}

	private static string Base64Encoding(byte[] input) {
		return Convert.ToBase64String(input);
	}

	private static byte[] Base64Decoding(String input) {
		return Convert.FromBase64String(input);
	}
}

File: LibsodiumSealedCryptoboxEncryptionString/LibsodiumSealedCryptoboxEncryptionString.dart
import 'dart:convert';
import 'dart:typed_data';
import 'package:flutter_sodium/flutter_sodium.dart';

main() {
/* add in pubspec.yaml:
dependencies:
  flutter_sodium: ^0.2.0
 */

  printC('Libsodium sealed crypto box hybrid string encryption');
  final plaintext = 'The quick brown fox jumps over the lazy dog';
  printC('plaintext: ' + plaintext);

  // encryption
  printC('\n* * * Encryption * * *');
  // for encryption you need the public key from the one you are sending the data to (party B)
  String publicKeyBBase64 =  'jVSuHjVH47PMbMaAxL5ziBS9/Z0HQK6TJLw9X3Jw6yg=';
  var publicKeyB = base64Decoding(publicKeyBBase64);
  String ciphertextBase64 = sealedCryptoboxEncryptionToBase64(publicKeyB, plaintext);
  printC('ciphertext (Base64): ' + ciphertextBase64);
  printC('output is (Base64) ciphertext');

  printC('\n* * * Decryption * * *');
  // for decryption you need your private key (privateKeyB) and your public key (publicKeyB)
  String privateKeyBBase64 = "yNmXR5tfBXA/uZjanND+IYgGXlrFnrdUiUXesI4fOlM=";
  var privateKeyB = base64Decoding(privateKeyBBase64);
  var ciphertextReceivedBase64 = ciphertextBase64;
  printC('ciphertext (Base64): ' + ciphertextReceivedBase64);
  printC('input is (Base64) ciphertext');
  var decryptedtext = sealedCryptoboxDecryptionFromBase64(privateKeyB, publicKeyB, ciphertextReceivedBase64);
  printC('plaintext:  ' + decryptedtext);
}

String sealedCryptoboxEncryptionToBase64(Uint8List publicKey, String plaintext) {
  var ciphertext = SealedBox.sealString(plaintext, publicKey);
  return base64Encoding(ciphertext);
}

String sealedCryptoboxDecryptionFromBase64(Uint8List privateKey, Uint8List publicKey, String data) {
  KeyPair keyPairB = new KeyPair(pk: publicKey, sk: privateKey);
  var deryptedtext = SealedBox.openString(base64Decoding(data), keyPairB);
  return deryptedtext;
}

Uint8List createUint8ListFromString(String s) {
  var ret = new Uint8List(s.length);
  for (var i = 0; i < s.length; i++) {
    ret[i] = s.codeUnitAt(i);
  }
  return ret;
}

String base64Encoding(Uint8List input) {
  return base64.encode(input);
}

Uint8List base64Decoding(String input) {
  return base64.decode(input);
}

void printC(String newString) {
  // for compatibility reasons to FlutterEmptyConsole
  print(newString);
}
